packages/react-components/src/components/InputBoxComponent.tsx

```diff
@@ -201,3 +634,818 @@ export const InputBoxButton = (props: InputBoxButtonProps): JSX.Element => {
+  const tags: TagData[] = []; // Tags passed back to the caller
+  const tagParseStack: TagData[] = []; // Local stack to use while parsing
+
+  let plainTextRepresentation = '';
+
+  let parseIndex = 0;
+  while (parseIndex < text.length) {
+    const foundHtmlTag = findNextHtmlTag(text, parseIndex);
+
+    if (!foundHtmlTag) {
+      if (parseIndex !== 0) {
+        // Add the remaining text to the plain text representation
+        plainTextRepresentation += text.substring(parseIndex);
+      } else {
+        plainTextRepresentation = text;
+      }
+      break;
+    }
+
+    if (foundHtmlTag.type === 'open' || foundHtmlTag.type === 'self-closing') {
+      const nextTag = parseOpenTag(foundHtmlTag.content, foundHtmlTag.startIdx);
+      // Add the plain text between the last tag and this one found
+      plainTextRepresentation += text.substring(parseIndex, foundHtmlTag.startIdx);
+      nextTag.plainTextBeginIndex = plainTextRepresentation.length;
+
+      if (foundHtmlTag.type === 'open') {
+        tagParseStack.push(nextTag);
+      } else {
+        nextTag.content = '';
+        nextTag.plainTextBeginIndex = plainTextRepresentation.length;
+        nextTag.plainTextEndIndex = plainTextRepresentation.length;
+        addTag(nextTag, tagParseStack, tags);
+      }
+    }
+
+    if (foundHtmlTag.type === 'close') {
+      const currentOpenTag = tagParseStack.pop();
+      const closeTagType = foundHtmlTag.content.substring(2, foundHtmlTag.content.length - 1).toLowerCase();
+
+      if (currentOpenTag && currentOpenTag.tagType === closeTagType) {
+        // Tag startIdx is absolute to the text. This is updated later to be relative to the parent tag
+        currentOpenTag.content = text.substring(
+          currentOpenTag.openTagIdx + currentOpenTag.openTagBody.length,
+          foundHtmlTag.startIdx
+        );
+
+        // Insert the plain text pieces for the sub tags
+        if (currentOpenTag.tagType === 'msft-mention') {
+          plainTextRepresentation =
+            plainTextRepresentation.slice(0, currentOpenTag.plainTextBeginIndex) +
+            trigger +
+            plainTextRepresentation.slice(currentOpenTag.plainTextBeginIndex);
+        }
+
+        if (!currentOpenTag.subTags) {
+          plainTextRepresentation += currentOpenTag.content;
+        } else if (currentOpenTag.subTags.length > 0) {
+          // Add text after the last tag
+          const lastSubTag = currentOpenTag.subTags[currentOpenTag.subTags.length - 1];
+          const startOfRemainingText =
+            (lastSubTag.closeTagIdx ?? lastSubTag.openTagIdx) + lastSubTag.tagType.length + 3;
+          const trailingText = currentOpenTag.content.substring(startOfRemainingText);
+          plainTextRepresentation += trailingText;
+        }
+
+        currentOpenTag.plainTextEndIndex = plainTextRepresentation.length;
+        addTag(currentOpenTag, tagParseStack, tags);
+      } else {
+        console.error(
```

Comment:
Throw errors instead of do console.error, otherwise it cannot be caught by contoso app, and cannot be hide by contoso

Reply:
Added to refactor task